tip-calculator/src/components/Button.test.tsx

```diff
@@ -0,0 +1,67 @@
+import { render, screen, fireEvent } from '@testing-library/react';
+import '@testing-library/jest-dom';
+
+import Button from './Button';
+import React from 'react';
+
+const mockOnClick = jest.fn();
+
+describe('Button component', () => {
+  it('Renders a Button with the correct text', () => {
+    render(
+      <Button selected={true} onClick={mockOnClick}>
+        Click Me
+      </Button>
+    );
+    const button = screen.getByText(/click me/i);
+    expect(button).toBeInTheDocument();
+  });
+
+  it('Allows us to click the button', () => {
+    render(
+      <Button selected={true} onClick={mockOnClick}>
+        Click Me
+      </Button>
+    );
+    const button = screen.getByText(/click me/i);
+    fireEvent.click(button);
+    expect(mockOnClick).toHaveBeenCalled();
+  });
+
+  it('Prevents button being clicked if disabled', () => {
+    render(
+      <Button selected={true} onClick={mockOnClick} disabled={true}>
+        Click Me
+      </Button>
+    );
+    const button = screen.getByText(/click me/i);
+    expect(button).toBeDisabled();
+  });
+
+  // it('Applies the correct styles when selected', () => {
```

Comment:
Styles can be enforced using snapshot tests

Reply:
I have added some snapshot tests for the button when selected, default and when disabled